### PR TITLE
Add ONNX rewriter

### PR DESCRIPTION
**Pull request by Giuseppe5**
Depends on #110

### CODE REVIEW

File: tests/brevitas/test_onnx_export.py [hunk 1]
```diff
@@ -145,9 +146,27 @@ def test_dynamic_quantization(
...
+                if node.op_type == "Gemm" or node.op_type == "MatMul":
+                    matmul_gemm_counter += 1
+
+                if node.op_type == "MatMulInteger":
+                    matmulinteger_counter += 1
+
+            # The number of Matmul+Gemm has to be less compared to the model pre-transformation
+            # This is not zero since there are matmul that are not linear layers so they are not replaced
+            # and some linears layers can be excluded from quantization
+            assert matmul_gemm_counter <= original_matmul_gemm_counter
```
Comment (Giuseppe5) [inline tests/brevitas/test_onnx_export.py:171]: A better test would have us monitoring exactly how many Matmul and Gemm we expect to have before/after transformations, and similarly with matmulinteger.

Considering all the other changes that we are doing plus new tests that we will be adding, maybe we could wait for this kind of implementation when everything is more stable.

**Comment by Giuseppe5** [comment]
It seems that `onnx_tool` has a bad typing which is making the tests to fail

**Comment by fxmarty** [comment]
python 3.8 is still largely used (see https://pypistats.org/packages/transformers), although EOL is in a few months. We thus probably don't want to have `python_requires=">3.9"` in setup.py, but should raise a meaningful error if the onnx_tool has issues with python 3.8

File: optimum/amd/brevitas/export.py [hunk 2]
```diff
@@ -1,15 +1,256 @@
...
+        )
+    return compute_graph
+
+
+def find_and_insert_matmulinteger(model_path):
+    print("Rewriting ONNX Graph with MatMulInteger ")
+    model_path = os.path.join(model_path, "model.onnx")
+    cfg = {"constant_folding": False, "node_rename": False, "if_fixed_branch": None, "fixed_topk": 0, "verbose": True}
+    original_output = onnx.load(model_path).graph.output
+    model = Model(model_path, cfg)
```
Comment (fxmarty) [inline optimum/amd/brevitas/export.py:216]: I am not familiar with onnx_tool `Model`, but isn't the model loaded twice here (doubling load time)? Can't we initialize a `Model` from a `ModelProto`, or alternatively retrieve `original_output` from the `Model`?

**Review by fxmarty — COMMENTED** [review]
Great work!

File: optimum/amd/brevitas/export.py (hunk 2 above)
Comment (Giuseppe5) [reply optimum/amd/brevitas/export.py:216]: Yeah you're right, it should work

**Review by fxmarty — APPROVED** [review]
LGTM

File: tests/brevitas/test_onnx_export.py (hunk 1 above)
Comment (mht-sharma) [inline tests/brevitas/test_onnx_export.py:171]:
```suggestion
            self.assertTrue(matmul_gemm_counter <= original_matmul_gemm_counter)
```

**Review by mht-sharma — APPROVED** [review]
Thanks! LGTM

**Pull request merged**